### PR TITLE
fix: fix shellecheck issues

### PR DESCRIPTION
**Pull request by Stebalien**
Apparently, export/local on the same line as shell substitution hides any errors from the shell substitution.

Also, lots of quotes.

### CODE REVIEW

File: build-go.sh [hunk 1]
```diff
@@ -173,12 +179,18 @@ function printInitialDistfile() {
...
+  "id": "$reponame",
+  "version": "$version",
+  "releaseLink": "$distname/$version",
+  "name": "$distname",
+  "owner": "$(< repo-owner)",
+  "description": "$(< description)",
+  "date": "$(date '+%B %d, %Y')",
+  "platforms": {}
+}
+EOF
```
Comment (hsanjuan) [inline build-go.sh]: :+1:

File: build-go.sh [hunk 2]
```diff
@@ -97,26 +101,28 @@ function doBuild() {
 	local output=$4
 	local version=$5
 
-	dir=$output
-	name=$(basename `pwd`)
-	binname=${name}_${version}_${goos}-${goarch}
+  local dir name binname
```
Comment (hsanjuan) [inline build-go.sh]: tab

File: build-go.sh [hunk 3]
```diff
@@ -202,21 +214,22 @@ function buildWithMatrix() {
 
 	mkdir -p "$output"
 
-	local distname=$(basename `pwd`)
+	local distname
+  distname=$(basename "$(pwd)")
```
Comment (hsanjuan) [inline build-go.sh]: tab

File: build-go.sh [hunk 4]
```diff
@@ -274,11 +287,12 @@ function buildSource() {
 	cp "$reporoot/$target" "$output"
 
 	# Calculate sha512 and write it to .sha512 file
-	shasum -a 512 $output/$target | awk "{gsub(\"${output}/\", \"\");print}" > $output/$target.sha512
-	local linksha512=$(cat $output/$target.sha512 | awk '{ print $1 }')
+  local linksha512 linkcid
```
Comment (hsanjuan) [inline build-go.sh]: tab

File: build-go.sh [hunk 5]
```diff
@@ -337,7 +352,8 @@ function startGoBuilds() {
 		fail "versions file $versions does not exist"
 	fi
 
-	export GOPATH=$(pwd)/gopath
+	export GOPATH
+  GOPATH="$(pwd)/gopath"
```
Comment (hsanjuan) [inline build-go.sh]: tab

File: build-go.sh [hunk 6]
```diff
@@ -378,9 +394,9 @@ function startGoBuilds() {
         go mod edit -require "$repo@$(git -C "$repopath" rev-parse HEAD)"
     fi
 
-    buildWithMatrix "$matfile" "$repo/$package" "$outputDir/$version" "$(currentSha $repopath)" "$version"
+    buildWithMatrix "$matfile" "$repo/$package" "$outputDir/$version" "$(currentSha "$repopath")" "$version"
 		echo ""
-	done < $versions
+	done < "$versions"
 
 	notice "build complete!"
```
Comment (hsanjuan) [inline build-go.sh]: indent seems wrong here

**Comment by Stebalien** [comment]
Tabs... I keep forgetting about that.

**Pull request merged**